core/src/main/java/com/oath/oak/OakMap.java

```diff
@@ -391,59 +426,470 @@ public boolean inBounds(K key) {
-     */
-    public <T> OakTransformView<K, T> createTransformView(Function<Map.Entry<ByteBuffer, ByteBuffer>, T> transformer){
-        return new OakTransformView<K, T>(internalOakMap,this, fromKey, toKey, transformer);
+
+    private OakIterator<OakRBuffer> valuesBufferIterator() {
+        return internalOakMap.valuesBufferViewIterator(fromKey, fromInclusive, toKey, toInclusive, isDescending);
     }
 
-    /* ---------------- Package visibility getters for the views methods -------------- */
-    MemoryManager getMemoryManager() { return memoryManager; }
+    private OakIterator<Map.Entry<ByteBuffer, OakRBuffer>> entriesBufferIterator() {
+        return internalOakMap.entriesBufferViewIterator(fromKey, fromInclusive, toKey, toInclusive, isDescending);
+    }
 
-    boolean getIsDescending(){
-        return isDescending;
+    /* ---------------- TODO: Move methods below to their proper place as they are implemented -------------- */
+
+
+    @Override
+    public void replaceAll(BiFunction<? super K, ? super V, ? extends V> function) {
+        throw new UnsupportedOperationException();
     }
 
-    boolean getFromInclusive() {
-        return fromInclusive;
+    @Override
+    public V computeIfAbsent(K key, Function<? super K, ? extends V> mappingFunction) {
+        return null;
     }
 
-    boolean getToInclusive() {
-        return toInclusive;
+    @Override
+    public V compute(K key, BiFunction<? super K, ? super V, ? extends V> remappingFunction) {
+        return null;
+    }
+
+    @Override
+    public V merge(K key, V value, BiFunction<? super V, ? super V, ? extends V> remappingFunction) {
+        return null;
+    }
+
+    @Override
+    public Entry<K, V> floorEntry(K key) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public K floorKey(K key) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Entry<K, V> ceilingEntry(K key) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public K ceilingKey(K key) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Entry<K, V> higherEntry(K key) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public K higherKey(K key) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Entry<K, V> firstEntry() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Entry<K, V> lastEntry() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Entry<K, V> pollFirstEntry() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Entry<K, V> pollLastEntry() {
+        throw new UnsupportedOperationException();
+    }
+
+    /* ---------------- View Classes -------------- */
+
+    static class KeySet<K> extends AbstractSet<K> implements NavigableSet<K> {
+
```

Comment:
is there a way to handle this with inheritance instead of composition?

Reply:
Could you elaborate? I'm not sure I understand what set you mean we should extend.